src/crypto.cpp

```diff
@@ -572,6 +572,153 @@ PublicKey::getLongId() const
 #endif
 }
 
+// Certificate Request
+
+static NameType
+typeFromGnuTLS(gnutls_x509_subject_alt_name_t type)
+{
+    switch(type) {
+    case GNUTLS_SAN_DNSNAME:
+        return NameType::DNS;
+    case GNUTLS_SAN_RFC822NAME:
+        return NameType::RFC822;
+    case GNUTLS_SAN_URI:
+        return NameType::URI;
+    case GNUTLS_SAN_IPADDRESS:
+        return NameType::IP;
+    default:
+        return NameType::UNKNOWN;
+    }
+}
+
+static gnutls_x509_subject_alt_name_t
+typeFromName(NameType type)
```

Comment:
`typeFromGnuTLS` <=> `GnuTLSFromName` maybe here?

The previous function is the invert operation, should be reflected into the API

Reply:
done